src/ed25519.rs

```diff
@@ -449,6 +431,26 @@ impl Keypair {
     }
 }
 
+impl Signer<Signature> for Keypair {
+    /// Sign a message with this keypair's secret key.
+    fn try_sign(&self, message: &[u8]) -> Result<Signature, SignatureError> {
```

Comment:
This trait provides an infallible `sign` method impl'd in terms of `try_sign`, which exists to support e.g. HSMs which can have communication errors while signing.

In the case of this library, since signing is always infallible, it just returns `Ok`